src/annotator/plugin/bucket-bar.js

```diff
@@ -0,0 +1,214 @@
+import $ from 'jquery';
+import { default as Delegator } from '../delegator';
+import scrollIntoView from 'scroll-into-view';
+import {
+  findClosestOffscreenAnchor,
+  constructPositionPoints,
+  buildBuckets,
+} from '../util/buckets';
+
+const BUCKET_SIZE = 16; // Regular bucket size
+const BUCKET_NAV_SIZE = BUCKET_SIZE + 6; // Bucket plus arrow (up/down)
+const BUCKET_TOP_THRESHOLD = 115 + BUCKET_NAV_SIZE; // Toolbar
+
+// Scroll to the next closest anchor off screen in the given direction.
+const scrollToClosest = function (anchors, direction) {
+  const closest = findClosestOffscreenAnchor(anchors, direction);
+  if (closest && closest.highlights?.length) {
+    scrollIntoView(closest.highlights[0]);
+  }
+};
+
+export default class BucketBar extends Delegator {
+  constructor(element, options, annotator) {
+    const defaultOptions = {
+      // gapSize parameter is used by the clustering algorithm
+      // If an annotation is farther then this gapSize from the next bucket
+      // then that annotation will not be merged into the bucket
+      // TODO: This is not currently used; reassess
+      gapSize: 60,
+      html: '<div class="annotator-bucket-bar"></div>',
+      // Selectors for the scrollable elements on the page
+      scrollables: ['body'],
+    };
+
+    const opts = { ...defaultOptions, ...options };
+    super($(opts.html), opts);
+
+    this.options = opts;
+    this.buckets = [];
```

Comment:
`buckets`, `index` and `tabs` were previously set on the prototype, but I can't see any reason they should be accessible/mutable.